src/moonstone-samples/lib/DataGridListSample.js

```diff
@@ -110,12 +148,70 @@ module.exports = kind({
 		this.modelIndex = idx;
 		return records;
 	},
+	generateDataGridList: function (itemType) {
+		if (this.$.gridList) {
+			this.$.gridList.destroy();
+		}
+
+		var props = utils.clone(this.dataListDefaults),
+			createdComponent;
+
+		switch (itemType) {
+		case 'HorizontalGridListImageItem':
+			props.minWidth = 600;
+			props.minHeight = 100;
+			props.components[0] = {
+				kind: HorizontalGridListImageItem,
+				bindings: [
+					{from: 'model.text', to: 'caption'},
+					{from: 'model.subText', to: 'subCaption'},
+					{from: 'model.url', to: 'source'},
+					{from: 'model.selected', to: 'selected', oneWay: false}
+				]
+			};
+			break;
+		case 'HorizontalGridListItem':
+			props.minWidth = 600;
+			props.minHeight = 100;
+			props.components[0] = {
+				kind: HorizontalGridListItem,
+				bindings: [
+					{from: 'model.text', to: 'caption'},
+					{from: 'model.subText', to: 'subCaption'},
+					{from: 'model.selected', to: 'selected', oneWay: false}
+				]
+			};
+			break;
+		}
+
+		createComponent = this.$.listPanel.createComponent(props, {owner: this});
+		createComponent.render();
```

Comment:
One last thing ... your local variable is `createdComponent` but you're assigning to `createComponent`